test/archive/har.d

```diff
@@ -0,0 +1,198 @@
+                return No.success;
+            }
+            // Maybe a warning if directory is not empty or something
+            if (verbose)
+                infoFile.writefln("output directory %s already exists", outputDir.formatDir);
+        }
+        else
+        {
+            if (verbose)
+                infoFile.writefln("mkdir %s", outputDir.formatDir);
+            mkdirRecurse(outputDir);
+        }
+    
+        auto lineRange = harFile.byLine(Yes.keepTerminator);
+        if (lineRange.empty)
+        {
+            infoFile.writefln("%s(1): Error: file is empty", harFilename);
+            return No.success;
+        }
+        string delimiter;
+        auto line = lineRange.front;
+        {
+            auto spaceIndex = line.indexOf(' ');
+            if (spaceIndex < 0)
+            {
+                infoFile.writefln("%s(1) Error: first line does not contain a space", harFilename);
+                return No.success;
+            }
+            delimiter = line[0 .. spaceIndex + 1].idup;
+        }
+    
+    LfileLoop:
+        for (;;)
+        {
+            // At this point, line starts with the delimiter
+            auto fileInfoString = line[delimiter.length .. $];
+    
+            // For now this will just be the filename
+            auto fileName = fileInfoString.stripRight;
+            {
+                auto dir = dirName(fileName);
+                if (dir.length > 0)
+                {
+                    auto fullDir = buildPath(outputDir, dir);
+                    if (!extractMkdir(fullDir))
+                        return No.success;
+                }
+            }
+            auto fullFileName = buildPath(outputDir, fileName);
+            infoFile.writefln("creating %s", fullFileName.formatFile);
+            extractedFilenames.put(fullFileName.idup);
+            {
+                auto currentOutputFile = File(fullFileName, "w");
+                scope(exit) currentOutputFile.close();
+                for (;;)
+                {
+                    lineRange.popFront();
+                    if (lineRange.empty)
+                        break LfileLoop;
+                    line = lineRange.front;
+                    if (line.startsWith(delimiter))
+                        break;
+                    currentOutputFile.write(line);
+                }
+            }
+        }
+        return Yes.success;
+    }
+}
+
+@property auto formatDir(const(char)[] dir)
+{
+    if (dir.length == 0)
+    {
+        dir = ".";
+    }
+    return formatQuotedIfSpaces(dir);
+}
+
+@property auto formatFile(const(char)[] file)
+  in { assert(file.length > 0); } do
+{
+    return formatQuotedIfSpaces(file);
+}
+
+/**
+Returns a formatter that will print the given string.  it will print
+it surrounded with quotes if the string contains any spaces.
+*/
+@property auto formatQuotedIfSpaces(T...)(T args) if (T.length > 0)
```

Comment:
inconsistent constant style. the if constraint is typically on a separate new line, see https://dlang.org/dstyle.html#phobos_declarations